src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import duke.command.Command;
+import duke.parser.Parser;
+import duke.storage.FileManager;
+import duke.storage.TaskRecords;
+import duke.ui.BotUI;
+
+/**
+ * Represents the class of the duke bot program.
+ * A <code>Duke</code> object consists of BotUI and TaskRecords.
+ */
+public class Duke {
+
+    private final BotUI ui;
+    private final TaskRecords taskList;
+
+    /**
+     * Constructs Duke object.
+     * The taskList is assigned to the previous duke chatBot data if the file is found.
+     * TaskRecords will be initialised to an empty taskList if file is not found.
+     */
+    public Duke() {
+        this.ui = new BotUI();
+        TaskRecords temp;
+        try {
+            temp = FileManager.read();
+        } catch (FileNotFoundException ex) {
+            temp = new TaskRecords();
+        }
+        this.taskList = temp;
+    }
+
+    /**
+     * Runs the duke chatBot program.
+     * Shows the text-based UI
+     */
+    void runBot() {
+        System.out.println(ui.sayHello());
+        boolean exitDuke = false;
+        while (!exitDuke) {
+            try {
+                System.out.print(BotUI.userSpeak());
+                Command c = Parser.parse(ui.readCommand());
+                System.out.println(c.execute(taskList, ui));
+                FileManager.write(this.taskList);
+                exitDuke = c.isExit();

```

Comment:
I like how the SLAP principle is followed here (and also in many other areas of the codebase). The code is very clear and readable.